### PR TITLE
Snabbvmx selftest

### PR DESCRIPTION
**Pull request by dpino**
Tests Ping, ARP and NDP resolution in Snabbvmx using a stock Ubuntu (the qemu.img that is used for other tests such as SnabbNFV).

I planned to add nexthop resolution tests here but I will put that in a different file. The tests also lacks VLAN tests counterpart which I will after nexthop tests.

The tcpreplay command is a custom script. Currently there's a PR to upstream to add similar functionality using packetblaster but while it's not merged I prefer to rely on this script.

```
sudo SNABB_PCI0=0000:81:00.0 SNABB_PCI1=0000:81:00.1 program/snabbvmx/tests/selftest.sh
Launch Snabbvmx
Waiting for VM listening on telnet port 5000 to get ready... [OK]
Ping to lwAFTR inet interface: OK
Ping to lwAFTR B4 side: OK
ARP request to lwAFTR: OK
NDP request to lwAFTR: OK
```

### CODE REVIEW

File: src/program/snabbvmx/tests/scripts/tcpreplay.lua [hunk 1]
```diff
@@ -0,0 +1,87 @@
+local S = require("syscall")
+local Tap = require("apps.tap.tap").Tap
+local lib = require("core.lib")
+local pcap = require("apps.pcap.pcap")
+local pci = require("lib.hardware.pci")
+
+function show_usage (code)
+   print(require("program.tcpreplay.README_inc"))
```
Comment (teknico) [inline src/program/snabbvmx/tests/scripts/tcpreplay.lua]: Cannot find the README file. Also, I'm not sure that location exists.

File: src/program/snabbvmx/tests/scripts/tcpreplay.lua [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+   return opts, unpack(args)
+end
+
+-- TODO: Duplicated function. Move to a common place.
+local function dir_exists (path)
+   local stat = S.stat(path)
+   return stat and stat.isdir
+end
+
+-- TODO: Duplicated function. Move to a common place.
```
Comment (teknico) [inline src/program/snabbvmx/tests/scripts/tcpreplay.lua]: It's good that there's a TODO, but do we really have to keep adding these duplicated functions? Wouldn't it be better to just remove the duplication once and for all?

File: src/program/snabbvmx/tests/scripts/tcpreplay.lua [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      config.link(c, "pcap.output -> nic.input")
+   end
+
+   engine.configure(c)
+   engine.main({duration=opts.duration})
+end
+
+-- Snabb shell cannot run a script that is a module, but it can run
+-- a Lua script. However in that case 'args' variable is not present.
+-- This function access directly to the command line argument list
```
Comment (teknico) [inline src/program/snabbvmx/tests/scripts/tcpreplay.lua]: "This function directly accesses the command line..."

File: src/program/snabbvmx/tests/selftest.sh [hunk 4]
```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env bash
```
Comment (teknico) [inline src/program/snabbvmx/tests/selftest.sh:1]: Gosh, three hundred more lines of shell script. Sigh.

**Comment by dpino** [comment]
Fixed the issues in tcpreplay script. Actually I didn't need most of the auxiliary functions so I remove them.

File: src/program/snabbvmx/tests/selftest.sh (hunk 4 above)
Comment (dpino) [reply src/program/snabbvmx/tests/selftest.sh:1]: I'm not a fan of bash either but it's what Snabb uses for tests, so not really a choice here.

File: src/program/snabbvmx/tests/scripts/tcpreplay.lua [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+   config.app(c, "pcap", pcap.PcapReader, filein)
+   config.app(c, "nic", Intel82599, { pciaddr = iface })
+   config.link(c, "pcap.output -> nic.rx")
+   engine.configure(c)
+   engine.main({duration = opts.duration, report={showlinks=true}})
+end
+
+-- Snabb shell cannot run a script that is a module, but it can run
+-- a Lua script. However in that case 'args' variable is not present.
+-- This function directly accesses to the command line argument list
```
Comment (teknico) [inline src/program/snabbvmx/tests/scripts/tcpreplay.lua]: Also remove the "to".

**Comment by teknico** [comment]
LGTM (with one last nit).

**Pull request merged**